math/binarymath/checkisnumberpoweroftwo.go

```diff
@@ -0,0 +1,37 @@
+// checkisnumberpoweroftwo.go
+// description: Is the number a power of two
+// details:
+// Checks if a number is a power of two- [Power of two](https://en.wikipedia.org/wiki/Power_of_two)
+// author(s) [red_byte](https://github.com/i-redbyte)
+// see checkisnumberpoweroftwo_test.go
+
+package binarymath
+
+import (
+	"math"
+)
+
+func IsPowerOfTwoBinaryMethod(x int) bool {
+	return x > 0 && (x&(x-1)) == 0
```

Comment:
Please add some explanatory comments as many people will not be familiar with how bit-wise operations work or how numbers are represented in binary.

An example of what you could write:
```go
// This function uses the fact that powers of 2 are represented
// like 10...0 in binary, and numbers one less than the power of 2
// are represented like 11...1.
// Therefore, using the and function:
//    10...0
//  & 01...1
//    00...0 -> 0
// This is also true for 0, which is not a power of 2, for which we
// have to add and extra condition.
```

Reply:
I also agree with @raklaptudirm on this 😄